build_posix/aclocal/options.m4

```diff
@@ -275,4 +275,21 @@ no)	wt_cv_crc32_hardware=no
 	AC_MSG_RESULT(no);;
 esac
 
+AC_MSG_CHECKING(if --enable-llvm option specified)
+AC_ARG_ENABLE(llvm,
+	[AS_HELP_STRING([--enable-llvm],
+	    [Configure with LLVM.])], r=$enableval, r=no)
+case "$r" in
+no)	wt_cv_enable_llvm=no;;
+*)	wt_cv_enable_llvm=yes;;
+esac
+AC_MSG_RESULT($wt_cv_enable_llvm)
+AM_CONDITIONAL([LLVM], [test x$wt_cv_enable_llvm = xyes])
+
+AC_MSG_CHECKING(if --with-llvm-config option specified)
+AC_ARG_WITH(llvm-config,
+	[AS_HELP_STRING([--with-llvm-config=DIR],
+	    [Path to LLVM config.])])
+AC_MSG_RESULT($with_llvm_config)
+
```

Comment:
Can you discuss why you went in this direction, rather than `AC_CHECK_PROG` and/or `AC_CHECK_HEADER`?

It seems like we could build if we find `llvm-config` in `$PATH` and/or some of the required header files?

Reply:
If that's the case, then you probably will need to version the test, that is, only build if `llvm-config --version` reports a sufficiently up-to-date version?

Reply:
As a simple approach, maybe it would be best to require `--enable-XXX` to do the build, but then use `$PATH` to find `llvm-config`? That way we don't build unless the user configures us to do so, but we use the compiler/binary/include-file configuration the user has set before running for the actual build.

Reply:
Understood. As discussed, I'll do it this way.

Reply:
491971a